File: src/net/dinkla/raytracer/samplers/Jittered.java
package net.dinkla.raytracer.samplers;

import net.dinkla.raytracer.math.Point2D;
import net.dinkla.raytracer.math.Random;

import java.util.List;

/**
 * Created by IntelliJ IDEA.
 * User: Jörn Dinkla
 * Date: 21.05.2010
 * Time: 18:13:36
 * To change this template use File | Settings | File Templates.
 */
public class Jittered extends Generator {

    public void generateSamples(int numSamples, int numSets, List<Point2D> samples) {
        int n = (int) Math.sqrt(numSamples);
        for (int p=0; p<numSets; p++) {
            for (int j=0; j<n; j++) {
               for (int k=0; k<n; k++) {
                   float x = (k + Random.randFloat()) / n;
                   float y = (j + Random.randFloat()) / n;
                   samples.add(new Point2D(x, y));
               }
            }
        }
    }

}

File: src/main/kotlin/net/dinkla/raytracer/samplers/Jittered.kt
package net.dinkla.raytracer.samplers

import net.dinkla.raytracer.math.Point2D
import net.dinkla.raytracer.math.Random

class Jittered : IGenerator {

    override fun generateSamples(numSamples: Int, numSets: Int, samples: MutableList<Point2D>) {
        val n = Math.sqrt(numSamples.toDouble()).toInt()
        for (p in 0 until numSets) {
            for (j in 0 until n) {
                for (k in 0 until n) {
                    val x = (k + Random.randFloat()) / n
                    val y = (j + Random.randFloat()) / n
                    samples.add(Point2D(x, y))
                }
            }
        }
    }

}
